R/showToast.R

```diff
@@ -25,26 +43,29 @@ showToast <- function(
   message,
   title = NULL,
   keepVisible = FALSE,
-  .options = list(
-    positionClass = "toast-bottom-center",
-    progressBar = TRUE,
-    timeOut = 4000,
-    closeButton = TRUE,
-    
-    # same as defaults
-    newestOnTop = FALSE,
-    preventDuplicates = FALSE,
-    showDuration = 300,
-    hideDuration = 1000,
-    extendedTimeOut = 1000,
-    showEasing = "swing",
-    hideEasing = "swing",
-    showMethod = "fadeIn",
-    hideMethod = "fadeOut"
-  ), 
+  .options = list(),
   session = shiny::getDefaultReactiveDomain()
 ) {
-
+  
+  # Incorrect option
+  if (!all(names(.options) %in% names(default_options))) {
+    stop('Incorrect option supplied', call. = FALSE)
+  }
+  
+  if (length(.options) == 0) {
```

Comment:
I think you can replace this entire if else with:

```
.options <- modifyList(default_options, .options)
```

Please test that the above works.

Reply:
Good to know. I knew there had to be a function that already did what I was doing, just couldn't find it. Thanks!

0a63e18300f81a98eb335b47353337de929f2046